cm_service_manager.py

```diff
@@ -0,0 +1,140 @@
+cluster_name = ''
+if args.cluster is not None:
+    cluster_name = args.cluster.lower()
+else:
+    cluster_name = select_from_items('cluster', cluster_names)
+
+for cluster in clusters:
+    if cluster.name.lower() == cluster_name:
+
+        # Get the service
+        services = cluster.get_all_services()
+        service_names = [s.name.lower() for s in services]
+        service_name = ''
+        if args.service is not None:
+            service_name = args.service.lower()
+        else:
+            service_name = select_from_items('service', service_names)
+
+        # Get the action to be performed on predefined actions
+        defined_actions = ['Service restart', 'Deploy Client Configuration', 'Rolling Restart']
+
+        for service in services:
+            if service.name.lower() == service_name:
+
+                action = select_from_items('action', [i.lower() for i in defined_actions])
+
+                if action == defined_actions[0].lower():  # Restart the service
+                    print "Executing Restart on service '%s' " % service.name.lower()
+                    command = service.restart().wait()
+
+                elif action == defined_actions[1].lower():    # Deploy client config
+                    try:
+                        print "Deploying Client Configurations on service '%s' " % service.name.lower()
+                        command = service.deploy_client_config().wait()
+                    except:
+                        # Since DCC unavailable in oozie, hue, zookeeper
+                        print "WARNING: Deploying Client configuration is not available for '%s' " % service.name.lower()
+                        sys.exit()
+
+                elif action == defined_actions[2].lower():
+                    print "Executing Rolling Restart on service '%s' " % service.name.lower()
+                    command = rolling_restart(service)
+
+
+        job_succeeded = command.success
+
+        message = "Subject: '%s' automation update \n\n" % action
+
+        message += "\nCluster : %s" % cluster_name
+        message += "\nService : %s" % service_name
+        message += "\nAction Performed: %s" % action
+
+        if job_succeeded:
+            print "%s is successful. Please check your inbox for details" % action
+            message += "\n\n\n %s is Successful" % action
+        else:
+            print "%s failed. Please check your inbox for details" % action
+            message += "\n\n\n%s Failed" % action
+
+        # Logic to send mail updates on the job status
+        try:
+            server = smtplib.SMTP(server, 25)    # SMTP server at port 25
+
+            from_email = 'anoop.mamgain@cerner.com'
+            server.sendmail(from_email, 'anoop.mamgain@cerner.com, pavan.bolar@cerner.com', message)    # Send mail
+            server.quit()
+        except:
+            print "Invalid email information. Please check SMTP server info or mailing address"
+
```

Comment:
Mail block added. Mails are being sent successfully but not the the DL.
I still need to modify and generalize this part. Currently its a bit hard coded.